src/App.tsx

```diff
@@ -1,56 +1,92 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
+import classNames from 'classnames';
+
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { PeopleList } from './components/PeopleList/PeopleList';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const delay = 1000;
+  const allPeople: Person[] = peopleFromServer;
+
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState<string>('');
+  const [applyQuery, setApplyQuery] = useState<string>('');
+
+  const filteredPeople = useMemo(() => {
+    return allPeople.filter(
+      person => person.name.toLowerCase().includes(query.toLowerCase().trim()),
+    );
+  }, [applyQuery, allPeople]);
+
+  const applySearch = useCallback(
+    debounce(setApplyQuery, delay),
+    [delay],
+  );
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applySearch(event.target.value);
+  };
+
+  const handleSelectPerson = (person: Person) => {
+    setQuery(person.name);
+    setApplyQuery(person.name);
+    setSelectedPerson(person);
+  };
```

Comment:
Wrap this in useCallback, so memo'ed `PeopleList` can avoid extra rerender when this function is passed